packages/server/test/react.test.tsx

```diff
@@ -389,6 +399,92 @@ test('mutation on mount + subscribe for it', async () => {
   });
 });
 
+describe('useMutation()', () => {
+  test('nullish input called with no input', async () => {
+    const { trpc, client } = factory;
+
+    function MyComponent() {
+      const allPostsQuery = trpc.useQuery(['allPosts']);
+      const deletePostsMutation = trpc.useMutation('deletePosts');
+
+      useEffect(() => {
+        allPostsQuery.refetch().then(async (allPosts) => {
+          expect(allPosts.data).toHaveLength(2);
+          await deletePostsMutation.mutateAsync();
+          const newAllPost = await allPostsQuery.refetch();
+          expect(newAllPost.data).toHaveLength(0);
+        });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+      }, []);
+
+      return <pre>{JSON.stringify(allPostsQuery.data ?? {}, null, 4)}</pre>;
+    }
+
+    function App() {
+      const [queryClient] = useState(() => new QueryClient());
+      return (
+        <trpc.Provider {...{ queryClient, client }}>
+          <QueryClientProvider client={queryClient}>
+            <MyComponent />
+          </QueryClientProvider>
+        </trpc.Provider>
+      );
+    }
+
+    const utils = render(<App />);
+    await waitFor(() => {
+      expect(utils.container).toHaveTextContent('first post');
+    });
+
+    await waitFor(() => {
+      expect(utils.container).toHaveTextContent('[]');
+    });
+  });
+
+  test('nullish input called with input', async () => {
+    const { trpc, client } = factory;
+
+    function MyComponent() {
+      const allPostsQuery = trpc.useQuery(['allPosts']);
+      const deletePostsMutation = trpc.useMutation('deletePosts');
+
+      useEffect(() => {
+        allPostsQuery.refetch().then(async (allPosts) => {
+          expect(allPosts.data).toHaveLength(2);
+          await deletePostsMutation.mutateAsync(['1']);
+          const newAllPost = await allPostsQuery.refetch();
+          expect(newAllPost.data).toHaveLength(1);
+        });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+      }, []);
+
+      return <pre>{JSON.stringify(allPostsQuery.data ?? {}, null, 4)}</pre>;
+    }
+
+    function App() {
+      const [queryClient] = useState(() => new QueryClient());
+      return (
+        <trpc.Provider {...{ queryClient, client }}>
+          <QueryClientProvider client={queryClient}>
+            <MyComponent />
+          </QueryClientProvider>
+        </trpc.Provider>
+      );
+    }
+
+    const utils = render(<App />);
+    await waitFor(() => {
+      expect(utils.container).toHaveTextContent('first post');
+      expect(utils.container).toHaveTextContent('second post');
+    });
+
+    await waitFor(() => {
+      expect(utils.container).not.toHaveTextContent('first post');
+      expect(utils.container).toHaveTextContent('second post');
+    });
+  });
```

Comment:
We should add a test that we're not breaking existing `mutate(null)`-functionality + check that docs are updated